src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,671 @@
+
+  /** A message to serve as placeholder. */
+  placeholder?: string
+
+  /** A dropdown can have a search field instead of trigger button. */
+  search?: boolean
+
+  /** The value in the edit text, if dropdown is a search. */
+  searchQuery?: string
+
+  /** A dropdown can have a toggle button. */
+  toggleButton?: boolean
+
+  /** The value of the dropdown. */
+  value?: DropdownListItem | DropdownListItem[]
+}
+
+export interface DropdownState {
+  value: DropdownListItem | DropdownListItem[]
+  backspaceDelete: boolean
+  focused: boolean
+  searchQuery?: string
+}
+
+export interface DropdownListItem {
+  header: string
+  key?: string
+  content?: string
+  image?: string
+}
+/**
+ * A Dropdown allows a user to select a value or a multitude of values from a number of options.
+ */
+export default class Dropdown extends AutoControlledComponent<
+  Extendable<DropdownProps>,
+  DropdownState
+> {
+  private inputRef: HTMLElement
+
+  static displayName = 'Dropdown'
+
+  static className = 'ui-dropdown'
+
+  static propTypes = {
+    ...commonUIComponentPropTypes,
+    defaultSearchQuery: PropTypes.string,
+    defaultValue: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+    fluid: PropTypes.bool,
+    getA11yStatusMessage: PropTypes.func,
+    getA11ySelectedMessage: PropTypes.func,
+    getA11yRemovedMessage: PropTypes.func,
+    getA11yRemoveItemMessage: PropTypes.func,
+    items: PropTypes.arrayOf(PropTypes.object),
+    itemToString: PropTypes.func,
+    multiple: PropTypes.bool,
+    noResultsMessage: PropTypes.string,
+    onBackspaceDelete: PropTypes.func,
+    onCloseIconClick: PropTypes.func,
+    onCloseIconKeyDown: PropTypes.func,
+    onContainerClick: PropTypes.func,
+    onDropdownChange: PropTypes.func,
+    onInputBlur: PropTypes.func,
+    onInputFocus: PropTypes.func,
+    onInputKeyDown: PropTypes.func,
+    onLabelClick: PropTypes.func,
+    onSearchQueryChange: PropTypes.func,
+    placeholder: PropTypes.string,
+    search: PropTypes.bool,
+    searchQuery: PropTypes.string,
+    toggleButton: PropTypes.bool,
+    value: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+  }
+
+  static defaultProps = {
+    as: 'div',
+  }
+
+  static autoControlledProps = ['searchQuery', 'value']
+
+  state: DropdownState = {
+    backspaceDelete: this.props.multiple ? true : undefined,
+    focused: false,
+    searchQuery: this.props.search ? '' : undefined,
+    value: this.props.multiple ? [] : null,
+  }
+
+  public renderComponent({
+    ElementType,
+    classes,
+    styles,
+    variables,
+    rest,
+  }: RenderResultConfig<DropdownProps>) {
+    const { search, multiple, toggleButton, getA11yStatusMessage, itemToString } = this.props
+    const { searchQuery } = this.state
+    const optionalDownshiftProps = {
+      // in multiple dropdown, we hold active values in the array, and default active is null.
+      ...(multiple && { selectedItem: null }),
+    }
```

Comment:
I think we can make it simpler, without ```...``` operation
```js
const optionalDownshiftProps = multiple && { selectedItem: null } || {}
```

Let me know what you think

Reply:
Probably a leftover. I had more than one param at some point. Will refactor, thank you :)